### ISSUE TITLE
Possible naming collisions with "require$$"

### ISSUE THREAD

**Issue by sebastian-lenz** [rollup/rollup-plugin-commonjs#238]
I just ran into an issue using rollup-plugin-commonjs when requiring other libraries that have been bundled with rollup. In my case, [React uses rollup-plugin-commonjs](https://github.com/facebook/react/blob/master/package.json#L80) and the bundled library contains the variable `require$$0`:

```js
// react/cjs/react.development.js, Line 15 and following
'use strict';

var objectAssign$1 = require('object-assign');
var require$$0 = require('fbjs/lib/warning');
var emptyObject = require('fbjs/lib/emptyObject');
var invariant = require('fbjs/lib/invariant');
```

This seems to generate a naming collision when requiring the library from another rollup project, in my generated bundle the part above is transformed to something like this:

```js
'use strict';

var objectAssign$1 = require$$0;
var require$$0 = warning_1;
var emptyObject = emptyObject_1;
var invariant = invariant_1;
```

The variable name `require$$0` is used twice and the resulting code does not work. The cause of the problem seems to be the function [`getRequired` in `src/transform.js`](https://github.com/rollup/rollup-plugin-commonjs/blob/master/src/transform.js#L94) which does not check if the picked name is already in use.

A highly simplified but quick fix for anyone running into this issue is to change the generated name here to something else or by adding some random string:

```js
if ( !name ) name = `require$$${Math.random().toString(36).substring(2, 15)}$${uid++}`;
```

**Comment by arcanis** [comment]
Should be fixed by #233 when it get merged

**Comment by sebastian-lenz** [comment]
Ah, sorry, I've must been blind. Same problem and same solution.
